### DISCUSSION TITLE
Wiki JS error Authentication LDAP

### DISCUSSION THREAD

**Discussion by Yuri-Zizeko** [requarks/wiki#3346] [Help / Questions]
Linux - Debian (linux) 10.2 x64
Node.js 10.15.2
PostgreSQL 12.1

Config:
LDAP URL: ldap://10.1.1.2:389
Admin Bind DN: cn=test,OU=MAN,OU=OLD,OU=Company,dc=test,dc=com
Admin Bind Credentials: P@$$w0rd!
Search Base: OU=MAN,OU=OLD,OU=Company,dc=test,dc=com
Search Filter: searchFilter: '(sAMAccountName={{username}})'

Error:
Invalid email / username or password.

Port 389 is open 
Checked by telnet

The domain controller responds

**Comment by Yuri-Zizeko** [comment]
Perhaps you need to install additional software?

**Comment by NGPixel** [comment]
In the admin area > Dev > Flags. Enable the LDAP Debug flag. Then look at the logs during a login attempt. You should see any error returned by the server.

**Reply by abnerrizzi** [reply]
I did that and nothing has changed in logs at leat in my docker container

**Reply by jko-nic** [reply]
It did not changed nothing even in case of Linux installation with SystemD service.

**Comment by Yuri-Zizeko** [comment]
2019-11-21T08:07:57.170Z [MASTER] warn: LDAP LOGIN ERROR (c1): Invalid email / username or password.

**Comment by Yuri-Zizeko** [comment]
This configuration works successfully on v1.0.117

**Comment by NGPixel** [comment]
Can you screenshot the Search Filter field in the admin area > Auth > LDAP?

**Comment by Yuri-Zizeko** [comment]
![1](https://user-images.githubusercontent.com/57997479/69516286-4b22b980-0f62-11ea-87ca-7a7c006e4b80.png)

**Comment by Yuri-Zizeko** [comment]
Does anyone work with Ldap authentication?

**Comment by kingfisher77** [comment]
Yes, we got it to work with ldaps.

**Comment by Yuri-Zizeko** [comment]
> Да, мы получили его для работы с ldaps.

Help with authentication settings please

**Comment by pirvana** [comment]
Help with the authentication settings, please, I need you to authenticate without having to modify the "mail" field, in the ldap
![image](https://user-images.githubusercontent.com/23135703/69878112-b93ce880-12a2-11ea-8206-bb5fb778bf8f.png)

**Comment by inginheiiro** [comment]
I've the same problem.... i believe there is a bug in file server/core/auth.js
in the  authenticate function "you" do not validate the defined authentication strategy, in this case LDAP... you are assuming jwt auth...

**Comment by NGPixel** [comment]
@inginheiiro No, you're confusing the authorization strategy (always JWT) and the authentication strategy (LDAP, local, google, etc.). They are completely unrelated.

**Reply by xth3usx** [reply]
Was the issue resolved? If so, how?

**Comment by kingfisher77** [comment]
> Help with the authentication settings, please, I need you to authenticate without having to modify the "mail" field, in the ldap
> ![image](https://user-images.githubusercontent.com/23135703/69878112-b93ce880-12a2-11ea-8206-bb5fb778bf8f.png)

You could try this:
`(|(uid=%uid)(mail=%uid))
`
Here LDAP attributes uid or mail can be used for wiki field uid.

**Comment by hbokh** [comment]
Had the same issue here. Using the exact same OpenLDAP credentails from wiki 1.0 in 2.0.

This helped: *enable self-registration in the LDAP module* - source: https://github.com/Requarks/wiki/issues/1232#issuecomment-555506212
At first I got redirected to the login-page.  As administrator (using the local login) added the new user to the right group and things worked as expected.

**Comment by pirvana** [comment]
@kingfisher77 Good friend, this is what appeared to me when placing what you requested, one more thing, it only worked without the symbol "|" (tube), any other option that could help me?
![image](https://user-images.githubusercontent.com/23135703/69956804-cb579a80-14df-11ea-81f6-d93e821e26d2.png)
![image](https://user-images.githubusercontent.com/23135703/69956831-ddd1d400-14df-11ea-907e-e88e776cee67.png)

**Comment by juliendms** [comment]
Our LDAP access is configured and working well. We had some troubles with some users who did not have the `mail` field filled in the AD. It seems like the `mail` is required for self-registration, maybe you should check if you can connect with an account from the LDAP that is using the same username AND mail. Are yo using self-registration ? Are the users already created on Wiki.JS ?

**Comment by pirvana** [comment]
Thank you for responding @juliendms , I comment that users are not created yet, I would like them to be created automatically, and I answer that if I complete the "mail" field in the LDAP if it works correctly, but if it does not have that field the user cannot enter, and I need it to be created automatically, or you can log in only with your user account without requesting "mail"

**Comment by juliendms** [comment]
I _would_ guess that for now the mail field is required when creating a user, therefore if this field is empty the creation of the user fails. It is _I guess_ because this field is the username when creating a new user locally.
You might be able to do a kind of script that will fill the mail with the username from the LDAP and append a `@company.com`, that's something worth checking. Sorry I cannot help much more.

**Reply by andresmorago** [reply]
this would be very interesting to try as some of the AD users dont have the "mail" field populated

**Comment by ethanmdavidson** [comment]
@pirvana it looks like you need to change the search filter to use the attribute that is actually holding the username. For example, my company uses Active Directory and I had to use the search filter `(sAMAccountName={{username}})` because the username is stored in the `sAMAccountName` attribute. 
Try using an LDAP browser to look at all the attributes on your user object, find the one that contains the username, and change the filter to use that attribute.

**Comment by vtjc2002** [comment]
Here is what worked for me and this works with "Allow self-registration"  as well

![wikijs_ldap_config](https://user-images.githubusercontent.com/8717495/70251047-d8c58c80-174c-11ea-8368-069df9e403e0.jpg)

**Comment by pirvana** [comment]
Thanks to all, really, I tried each one of the solutions that they wrote to me but still it doesn't let me just talk to the user, without having the "mail" field, configured in the LDAP, it would be great if this did not request :(, but Thank you all, I will continue looking for some way to solve this

**Comment by Yuri-Zizeko** [comment]
000004DC: LdapErr: DSID-0C090A4C, comment: In order to perform this operation a successful bind must be completed on the connection., data 0, v3839

**Comment by dieterv** [comment]
I've been hitting the exact same "warn: LDAP LOGIN ERROR (c1): Invalid email / username or password." error for a couple of days now.

Finally solved it after:
*  resetting the password
*  setting the "user cannot change password" property
*  setting the "password never expires" property

of the Active Directory "Admin Bind DN" account.
I'm sure the exact same password was used when creating the "Admin Bind DN" account so I have no idea why this worked...

**Comment by fordanic** [comment]
Using the same setup as for my 1.0 installation but this fails for my 2.0 installation with the following error:
` warn: LDAP LOGIN ERROR (c1): Invalid email / username or password.`
Used credentials are good as they work for other services using a similar LDAP connection.

Running a docker setup of wikijs. What can I do to improve logging?

**Comment by fordanic** [comment]
To me, I in the end solved it by removing all "" that were used for bindings and similar.

**Comment by julianoduarte1983** [comment]
Hello guys,

   I was with the same problem. 

   So, I solved it enabling auto self-registration on LDAP authentication.

   Thanks @hbokh  ;)

**Comment by SirGibihm** [comment]
Keep in mind that the same error message (see below) comes up when your Bind Credentials **AND/OR** your Login Credentials are wrong. The logging from my docker container (wikijs version 2.2.51, LDAP Logging enabled through the debug flag at Administration>Developer Tools>Flags) show for both invalid login credentials the following error: 
```bash
warn: LDAP LOGIN ERROR (c1): Invalid email / username or password.
```

In other news: My LDAPS example config that works perfectly for Wikijs 2 looks like this, I had to redact passwords etc. but I hope it helps you with syntax; everything blurred out is a normal string: 

<img width="1005" alt="Screenshot 2020-04-07 at 13 58 31" src="https://user-images.githubusercontent.com/61882185/78667256-e779a700-78d8-11ea-8f81-c27da12771d8.png">

**Comment by denisbonate** [comment]
Help me please.


![image](https://user-images.githubusercontent.com/57956632/79245818-fae3bf80-7e4e-11ea-8ca1-448af4553033.png)

**Comment by ramonjoaquim** [comment]
Try
` (sAMAccountName={{username}}*)`
and verify if your AD have the mail mapped field for e-mail, in my case I discovered the mail field are **userPrincipalName**, then my search filter looks like that
Solution
` (|(sAMAccountName={{username}}*) (userPrincipalName={{username}}*))`
![image](https://user-images.githubusercontent.com/39497210/80383123-8d398980-8879-11ea-862c-e71bc4d39f3f.png)

**Reply by abnerrizzi** [reply]
this solved all my issues

an other option it is check self-registration.

**Comment by flagal** [comment]
LDAP works for me. The problem was  **Search Filter** field. I put in (mail={{username}}). In **Unique ID Field Mapping**, I put sAMAccoutName.

**Comment by Solotovski** [comment]
LDAP works for me. You should set a email address on LDAP server.

**Reply by dkyrgia** [reply]
Hi Solotovski, If I am not asking much, can you please post a screenshot of your settings for the ldap connection? I am trying for two days now and I cannot figure what's wrong

**Reply by abnerrizzi** [reply]
![image](https://github.com/requarks/wiki/assets/2419599/b664eb08-dce6-462c-84d7-11ce875bd6f8)

**Comment by slicedmass** [comment]
For those still having issues, since the error messages are pretty unhelpful. Make sure when you are specifying the Admin Bind DN/Search Base that you add quotes around entries with spaces. For example:

OU=contoso users,OU=contoso company,DC=contoso,DC=com

would become

OU="contoso users",OU="contoso company",DC=contoso,DC=com
